### PR TITLE
[API] teamSearch show teams with no members if user is admin

### LINKED ISSUE #21176: API regression: teamSearch does not show teams with no members
https://gitea.com/api/swagger#/organization/teamSearch

that's why https://gitea.com/gitea/go-sdk/pulls/604 still fails
(https://drone.gitea.com/gitea/go-sdk/602/1/3)

### PR DESCRIPTION
**Pull request by 6543**
close #21176

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: tests/integration/api_org_test.go [hunk 1]
```diff
@@ -151,3 +152,38 @@ func TestAPIGetAll(t *testing.T) {
...
+		req = NewRequest(t, "GET", fmt.Sprintf("/api/v1/orgs/%s/teams/search?q=%s&token=%s", orgName, "lehr", token))
+		resp := MakeRequest(t, req, http.StatusOK)
+		data := struct {
+			Ok   bool
+			Data []*api.Team
+		}{}
+		DecodeJSON(t, resp, &data)
+		assert.True(t, data.Ok)
+		if assert.Len(t, data.Data, 1) {
+			assert.EqualValues(t, "Lehr", data.Data[0].Name)
```
Comment (KN4CK3R) [inline tests/integration/api_org_test.go]:
```suggestion
			Name:                    "Empty",
			IncludesAllRepositories: true,
			Permission:              "read",
			Units:                   []string{"repo.code", "repo.issues", "repo.ext_issues", "repo.wiki", "repo.pulls"},
		})
		MakeRequest(t, req, http.StatusCreated)

		// search for team with no member
		req = NewRequest(t, "GET", fmt.Sprintf("/api/v1/orgs/%s/teams/search?q=%s&token=%s", orgName, "empty", token))
		resp := MakeRequest(t, req, http.StatusOK)
		data := struct {
			Ok   bool
			Data []*api.Team
		}{}
		DecodeJSON(t, resp, &data)
		assert.True(t, data.Ok)
		if assert.Len(t, data.Data, 1) {
			assert.EqualValues(t, "Empty", data.Data[0].Name)
```
or `Leer` but not `Lehr`

Comment (6543) [reply tests/integration/api_org_test.go]: oh common ...

I put it by purpose as a smal "easter egg" as any string would have done it ;)

Comment (6543) [reply tests/integration/api_org_test.go]: same as case insenstive check

Comment (KN4CK3R) [reply tests/integration/api_org_test.go]: That would have been a good excuse at school 👍 "Teacher, that's just an easter egg"

Comment (6543) [reply tests/integration/api_org_test.go]: now that it has been discovered I'll move it to normal :/

File: tests/integration/api_org_test.go [hunk 2]
```diff
@@ -151,3 +152,38 @@ func TestAPIGetAll(t *testing.T) {
...
+		// create team with no member
+		req = NewRequestWithJSON(t, "POST", fmt.Sprintf("/api/v1/orgs/%s/teams?token=%s", orgName, token), &api.CreateTeamOption{
+			Name:                    "Empty",
+			IncludesAllRepositories: true,
+			Permission:              "read",
+			Units:                   []string{"repo.code", "repo.issues", "repo.ext_issues", "repo.wiki", "repo.pulls"},
+		})
+		MakeRequest(t, req, http.StatusCreated)
+
+		// search for team with no member
```
Comment (6543) [inline tests/integration/api_org_test.go]:
```suggestion
		// case-insensitive search for teams that have no members
```

File: models/organization/team.go [hunk 3]
```diff
@@ -123,35 +123,20 @@ func (opts *SearchTeamOptions) toCond() builder.Cond {
...
-
-	if opts.UserID > 0 {
-		sess = sess.Join("INNER", "team_user", "team_user.team_id = team.id")
-	}
-
-	if opts.PageSize == -1 {
-		opts.PageSize = int(count)
-	} else {
-		sess = sess.Limit(opts.PageSize, (opts.Page-1)*opts.PageSize)
+	if opts.PageSize != -1 {
```
Comment (KN4CK3R) [inline models/organization/team.go]: Isn't that always true and can therefore be removed?

Comment (6543) [reply models/organization/team.go]: as models/** func are our internal datastore api ... I would let this in as it is common behavior ...

we might refactor it, so SetDefaultValues does respect this by default but, that needs regression testing and wont be scope of this pull

Comment (6543) [reply models/organization/team.go]: > Isn't that always true and can therefore be removed?

I didn't go throu all func's calling this func - so I cant say for sure atm, but that does not matter

Comment (KN4CK3R) [reply models/organization/team.go]: I have seen 4 calls which all provide the `ListOptions` and even if it's not set the default value is 0.

Comment (6543) [reply models/organization/team.go]: well I dont mind to enforce it for now until one needs a non paginated one

**Review by KN4CK3R — APPROVED** [review]

**Pull request merged**